lib/v2/screens/sign_up/claim_invite/components/bottom_container.dart

```diff
@@ -20,15 +19,15 @@ class BottomContainer extends StatefulWidget {
 }
 
 class _BottomContainerState extends State<BottomContainer> {
-  late SignupBloc _signupBloc;
+  late SignupBloc _bloc;
   final _keyController = TextEditingController();
   final Debouncer _debouncer = Debouncer(milliseconds: 600);
 
   @override
   void initState() {
     super.initState();
-    _signupBloc = BlocProvider.of<SignupBloc>(context);
-    _keyController.text = '';
+    _bloc = BlocProvider.of<SignupBloc>(context);
+    _keyController.text = _bloc.state.claimInviteState.inviteMnemonic ?? '';
     _keyController.addListener(_onInviteCodeChanged);
   }
```

Comment:
These are from the past PR but it needs to change.

This fragment with the listener do you need listen **every state change**? because the main idea is only execute the listener once a condition is met.
```
    return BlocListener<SignupBloc, SignupState>(
      listenWhen: (previousState, currentState) => previousState != currentState,
      listener: (context, state) {
        if (state.claimInviteState.pageState == PageState.success &&
            !state.claimInviteState.inviteMnemonic.isNullOrEmpty) {
          _keyController.text = state.claimInviteState.inviteMnemonic!;
        }
      },
```
      
Instead try to use the pageCommand pattern that Gery bring and also do the logic into the bloc.
Example:
```
    return BlocListener<SignupBloc, SignupState>(
      listenWhen: (_, currentState) => currentState.pageCommand != null,
      listener: (_, state) {
        if (state.pageCommand is SetInitialInviteCode) {
          _keyController.text = state.claimInviteState.inviteMnemonic!;
        }
      },
```
Finally this BuildWhen condition it is basically the default behavior of the BlocBuilder **to rebuild on every state change** so there is no point in this line you can remove it.
```
      child: BlocBuilder<SignupBloc, SignupState>(
        buildWhen: (previousState, currentState) => previousState != currentState,
```